packages/core/src/factory.ts

```diff
@@ -37,15 +39,31 @@ export function factory<Resource, Input, Output, Config, LoadOpts>(
         // build the config for this request
         const configMerger = configMergerFactory(defaultConfig, customConfig, c);
         const configObj: Config | undefined = configMerger(input);
+        let inputValidations: IPrismDiagnostic[] = [];
 
         // find the correct resource
         let resource: Resource | undefined;
         if (components.router) {
-          resource = components.router.route({ resources, input, config: configObj }, defaultComponents.router);
+          try {
+            resource = components.router.route({ resources, input, config: configObj }, defaultComponents.router);
+          } catch (error) {
+            const { message, name, status } = error as ProblemJsonError;
+            // rethrow error we if we're attempting to mock
+            if ((configObj as IPrismConfig).mock) {
```

Comment:
Hmm, I don't get it.

> Shouldn't we check that it's a server mismatch error?

If **mock is true** we should re-throw **all** errors (server mismatch, path mismatch, method mismatch whatever).

If **mock is false** (aka. forwarding is turned on) don't throw errors but put the on the input validation errors list.

> Otherwise this won't throw the "route not matched" in case mock is enabled

Wait, isn't my code doing the exact opposite? If mock is enabled it **will** throw "route not matched"

Reply:
Yeah, I just didn't read it correctly